PresentScreenings/ScreeningsPlanner.cs

```diff
@@ -0,0 +1,119 @@
+                    {
+                        screening.ToggleFilmFanAttendance(filmFan);
+                        _controller.UpdateAttendanceStatus(screening);
+                        _plannedScreenings.Add(screening);
+                    }
+                    _controller.ReloadScreeningsView();
+                }
+
+                // Display the results of this rating.
+                DisplayResultsOfRating(rating, filmFan, films);
+
+                // Iterate to next lower rating.
+                rating.Decrease();
+            }
+
+            // Display the sceenings that were planned in this session.
+            DisplayPlannedScreenings();
+
+            // Call the display function with the result string.
+            _displayResults(_builder.ToString());
+        }
+        #endregion
+
+        #region Private Methods
+        private bool IsPlannable(Screening screening, List<Film> films)
+        {
+            var inSelectedFilms = films.Count(f => f.FilmId == screening.FilmId) > 0;
+            return screening.HasPlannableStatus() && inSelectedFilms;
+        }
+
+        private bool HasAttendedScreening(Film film, string filmFan)
+        {
+            return ViewController.FilmScreenings(film.FilmId).Any(s => s.FilmFanAttends(filmFan));
+        }
+
+        private void DisplayResultsOfRating(FilmRating rating, string filmFan, List<Film> films)
+        {
+            // Display summary for this rating.
+            int highRatedFilmCount = films.Count;
+            int plannedFilmCount = films.Where(f => HasAttendedScreening(f, filmFan)).Count();
+            string fmt = "{0} out of {1} films with rating {2} planned for {3}.";
+            _builder.AppendFormat(fmt, plannedFilmCount, highRatedFilmCount, rating, filmFan);
+
+            // Display films with this rating that weren't planned.
+            if (highRatedFilmCount - plannedFilmCount > 0)
+            {
+                string unplannedFmt = "Films with rating {0} that could not be planned for {1}:";
+                _builder.AppendLine();
+                _builder.AppendFormat(unplannedFmt, rating, filmFan);
+                _builder.AppendLine();
+                var unplannedFilms = films.Where(f => !HasAttendedScreening(f, filmFan));
+                _builder.AppendJoin(Environment.NewLine, unplannedFilms);
+                _builder.AppendLine();
+            }
+            _builder.AppendLine();
+        }
+
+        private void DisplayPlannedScreenings()
+        {
+            // Display newly planned screenings.
+            _builder.AppendLine();
+            if (_plannedScreenings.Count > 0)
+            {
+                _builder.AppendFormat("{0} screenings planned:", _plannedScreenings.Count);
+                _builder.AppendLine();
+                _plannedScreenings.Sort();
+                _builder.AppendJoin(Environment.NewLine, _plannedScreenings.Select(s => s.ToPlannedScreeningString()));
```

Comment:
Please consider formatting this in fixed-width columns to display is like a table.
Only if you have spare time! This text stuff will all be replaced by controls (I suppose).

Reply:
No done (no spare time).